projects/core/components/dialog/dialog.factory.ts

```diff
@@ -41,17 +42,23 @@ type ExtractDialogResult<T, K extends keyof T = ContextKeys<T>> = [K] extends [n
         ? R
         : void;
 
+type Options<T> = Omit<TuiDialogOptions<T>, 'data'> & {injector: Injector};
+
 export function tuiDialog<
     T,
     K extends ContextKeys<T>,
     D extends ExtractDialogData<T, K>,
     R extends ExtractDialogResult<T, K>,
 >(
     component: AssertNotMultipleContexts<T, K>,
-    options?: Partial<Omit<TuiDialogOptions<D>, 'data'>>,
+    {injector, ...options}: Partial<Options<D>> = {},
 ): (data: D) => Observable<R> {
-    const dialogService = inject(TuiDialogService);
-    const injector = inject(INJECTOR);
+    if (!injector) {
+        assertInInjectionContext(tuiDialog);
```

Comment:
Do we need it? Isn't it already inside `inject` method?

Reply:
```ts
ngDevMode && assertInInjectionContext(tuiDialog);
```

Reply:
No, no one does that

Reply:
I figure out, but does we really need `readable error` in production instead of development mode?

Reply:
This is not a warning, this is an assertion, so the prod should have this error.
But readability on prod will be worse, due to minification of the code.
But I still don't see the point in complicating the condition here.